var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -337,21 +336,13 @@ def bootstrap_args(self):
         if spec.satisfies("%intel@:2021.6.0"):
             args.append("CXXFLAGS=-diag-disable=2196")
 
-        if self.spec.satisfies("platform=windows"):
-            args.append("-GNinja")
-            self.generator = ninja
+        # always use ninja
+        args.append("--generator=Ninja")
```

Comment:
This will fail on Windows, you still need the conditional. 
`--generator` is only valid for bootstrapping on Unix as it uses a shell script, but on Windows we use an older version of CMake, and that only takes the `-G` argument.